bundles/binding/org.openhab.binding.panasonictv/src/main/java/org/openhab/binding/panasonictv/internal/PanasonicTVBinding.java

```diff
@@ -0,0 +1,238 @@
+		logger.debug("internalReceiveCommand() for item: " + itemName
+				+ " with command: " + command.toString());
+
+		if (this.providers.isEmpty()) {
+			logger.error("Binding is properly configured or loaded. No provider was found.");
+			return;
+		}
+
+		for (PanasonicTVBindingProvider provider : this.providers) {
+			PanasonicTVBindingConfig config = provider
+					.getBindingConfigForItem(itemName);
+			if (config == null)
+				continue;
+			int response = sendCommand(config);
+			if (response != 200) {
+				logger.warn("Command " + config.getCommand()
+						+ " to TV with IP " + registeredTVs.get(config.getTv())
+						+ " failed with HTTP Reponse Code " + response);
+				continue;
+			}
+			eventPublisher.postUpdate(itemName, OnOffType.OFF);
+		}
+
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	protected void internalReceiveUpdate(String itemName, State newState) {
+		logger.debug("internalReceiveUpdate() for item: " + itemName
+				+ " with state: " + newState.toString());
+	}
+
+	/**
+	 * @{inheritDoc
+	 */
+	@Override
+	public void updated(Dictionary<String, ?> config)
+			throws ConfigurationException {
+
+		if (config != null) {
+
+			// to override the default refresh interval one has to add a
+			// parameter to openhab.cfg like
+			// <bindingName>:refresh=<intervalInMs>
+			String refreshIntervalString = (String) config.get("refresh");
+			if (StringUtils.isNotBlank(refreshIntervalString)) {
+				refreshInterval = Long.parseLong(refreshIntervalString);
+			}
+
+			for (Enumeration<?> e = config.keys(); e.hasMoreElements();) {
+				String tv = (String) e.nextElement();
+				if (tv.equalsIgnoreCase("service.pid")
+						|| tv.equalsIgnoreCase("refresh"))
+					continue;
+				logger.info("TV registered '" + tv + "' with IP '"
+						+ config.get(tv) + "'");
+				registeredTVs.put(tv, config.get(tv).toString());
+			}
+
+			if (registeredTVs.isEmpty()) {
+				setProperlyConfigured(true);
+				logger.debug("No TV was registered in config file");
+			} else
+				setProperlyConfigured(true);
+		}
+	}
+
+	/**
+	 * @author André Heuer
+	 *
+	 * This methods sends the command to the TV
+	 * 
+	 * @return HTTP response code from the TV (should be 200)
+	 */
+	int sendCommand(PanasonicTVBindingConfig config) {
+		String command = config.getCommand().toUpperCase();
+
+		String soaprequest = "";
+
+		if (config.getCommand().toUpperCase().startsWith("HDMI"))
+			soaprequest = "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\"><s:Body><u:X_SendKey xmlns:u=\"urn:panasonic-com:service:p00NetworkControl:1\"><X_KeyEvent>NRC_"
+					+ command
+					+ "</X_KeyEvent></u:X_SendKey></s:Body></s:Envelope>\r";
+		else
+			soaprequest = "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\"><s:Body><u:X_SendKey xmlns:u=\"urn:panasonic-com:service:p00NetworkControl:1\"><X_KeyEvent>NRC_"
+					+ command
+					+ "-ONOFF</X_KeyEvent></u:X_SendKey></s:Body></s:Envelope>\r";
```

Comment:
i would prefer providing this request as a static String using "String.format" with placeholders